sympy/solvers/tests/test_inequalities.py

```diff
@@ -390,10 +390,11 @@ def test_issue_10198():
 def test_issue_10047():
     # issue 10047: this must remain an inequality, not True, since if x
     # is not real the inequality is invalid
-    # assert solve(sin(x) < 2) == (x <= oo)
+    assert solve(sin(x) < 2) == (x <= oo)
+    assert solveset(sin(x) < 2, domain=S.Reals) == S.Reals
```

Comment:
What has changed here?

Reply:
The original test had that line commented so that it passed with the original behavior being that `x <= oo` would be changed to true. That is not correct unless we know x is real. This tests that the solution is specified by `x <= oo` without it being default true. The second assertion is also part of issue #10047, so I added it also

Reply:
I ended up changing it back to the original test (although I left the second test that I added there), so that it passes with the current implementation of inequality solvers. Only because that issue is beyond the scope of this PR. See issue #17840